jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -215,6 +224,42 @@ public Entry acquire()
         return null;
     }
 
+    /**
+     * Utility method to acquire an entry from the pool,
+     * reserving and creating a new entry if necessary.
+     *
+     * @param creator a function to create the pooled value for a reserved entry.
+     * @return an entry from the pool or null if none is available.
+     */
+    public Entry acquire(Function<Entry, T> creator)
```

Comment:
There are no tests for this new method.

Furthermore, why does it take an `Entry` as parameter? I think it should just be a `Supplier<T>` to create the object as the creator is called within the acquisition process, so the implementation does all the plumbing with the `Entry`.

Reply:
It takes an entry because it is typical (both our usages) for a two way association to be made between the pooled item and its entry.  This allows the item to pass the entry in the constructor or associate it with an attachment etc.

I'll add some tests

Reply:
I don't agree.  There is currently no dependency on `Attachable` from Pool and nothing that says an `Attachable` must be used for the association... or even if something is `Attachable` that it may be so for another reason.

Reply:
Note also that with the `Reservation` approach, this method is not so important as the API is clear and it is probably less faff to not use this method.   So if we don't really agree on this method, it is OK to just not have it... so long as we have `Reservation` or something like it.

Reply:
With the Reservation API, it is no longer difficult to call the API directly, so this method is no longer needed.